portfolio/src/main/java/com/google/sps/data/BlogMessage.java

```diff
@@ -23,16 +23,40 @@
   private final String sender;
   private final long timestamp;
   private final String image;
-  private final String tags;
+  private final String tag;
   private static ArrayList<String> messageReplies;
 
-  public BlogMessage(long id, String tags, String message, String image, String sender, ArrayList<String> messageReplies, long timestamp) {
+  public BlogMessage(long id, String tag, String message, String image, String sender, ArrayList<String> messageReplies, long timestamp) {
```

Comment:
I would keep this in its own file since it is used in other parts of the code as well, and the uses of BlogMessage aren't dependent on your hashtable API. For example, storing a BlogMessage in Datastore, doesn't also require that Datastore knows about the getMessage function.